src/thing_physics.c

```diff
@@ -662,6 +843,28 @@ TbBool thing_is_exempt_from_z_axis_clipping(const struct Thing *thing)
     }
     return false;
 }
+
+unsigned short get_slide_z_coord(const struct Thing *thing, const struct Coord3d *pos)
+// function at 451700. Original name unknown.
+{
+  unsigned short clipbox_size = thing->clipbox_size_yz;
+  long height = get_ceiling_height_above_thing_at(thing, pos);
+  short z_thing = (short)thing->mappos.z.val;
+  short z_pos = (short)pos->z.val;
+  if ( (height - 1) <= (z_pos + clipbox_size) )
+  {
+    return (height - clipbox_size) - 1;
+  }
+  if ( z_pos == z_thing )
+  {
+    return pos->z.val;
+  }
+  if ( z_pos < z_thing )
+  {
+    return (z_pos & 0xFF00) + 256;
```

Comment:
This is a weird decompiled pattern. What is happening here actually?
It looks like "if thing fell through floor it should be moved somehow"

Reply:
I don't actually know; I just copied the decompiled code.